examples/dataflow-flex-python/df-sample-xml-to-bq/Dockerfile

```diff
@@ -0,0 +1,16 @@
+# This Dockerfile is to create an Image used by launcher to
+# launch Dataflow workers.
+
+FROM gcr.io/dataflow-templates-base/python38-template-launcher-base:20220418_RC00
+
+ARG WORKDIR=/dataflow/template
+
+RUN mkdir -p ${WORKDIR}
+
+WORKDIR ${WORKDIR}
+
+COPY xmltobq.py .
+
+ENV FLEX_TEMPLATE_PYTHON_PY_FILE="${WORKDIR}/xmltobq.py"
+
+RUN pip install apache-beam[gcp]==2.40.0 xmltodict==0.13.0
```

Comment:
Why isn't this similar to the boilerplate Dockerfile produced by `gcloud dataflow flex-template build`?
That would normally produce a Dockerfile with something like:
```
FROM {BASE_IMAGE}

ENV FLEX_TEMPLATE_PYTHON_PY_FILE=/template/{PY_FILE}
ENV FLEX_TEMPLATE_PYTHON_REQUIREMENTS_FILE=/template/{REQUIREMENTS_FILE}

COPY {PY_PATHS} /template/

RUN apt-get update && apt-get install -y libffi-dev git && rm -rf /var/lib/apt/lists/* && pip install --no-cache-dir -U -r /template/{REQUIREMENTS_FILE}
```

Since there's nothing custom being added in this Dockerfile, why not just let `gcloud dataflow flex-template build` build the Dockerfile contents instead and remove this file altogether?

Reply:
I think when we pass FLEX_TEMPLATE_PYTHON_REQUIREMENTS_FILE, launcher will try to download and install the dependencies, and since i am using some custom packages like models my file is customised. I am also using a released stable launcher image as base image. Also i wanted to use python 3.8 in launcher and SDK both.